components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIManagerConfiguration.java

```diff
@@ -1773,12 +1774,42 @@ private void setEventHubConfiguration(OMElement omElement) {
                 if (eventTypeElement != null) {
                     eventHubPublisherConfiguration.setType(eventTypeElement.getText().trim());
                 }
+                if (Boolean.parseBoolean(System.getenv("FEATURE_FLAG_REPLACE_EVENT_HUB"))) {
```

Comment:
can't we set this by default without having a flag

Reply:
This is a flag to indicate that this feature in progress. Unless someone specifically enables it, nobody sees the feature.